### ISSUE TITLE
scaffold _s not creating the theme but saying it did

### ISSUE THREAD

**Issue by yivi** [wp-cli/wp-cli#1698]
I run ` wp scaffold _s test_theme --theme_name="Test Theme" --author="Test Author" --author_uri="https://example.es/"`, I get `Success: Created theme 'Test Theme'.`, but no theme files are created anywhere I see.
My file structure is this:
```
- root
--- wordpress
--- content
----- themes
----- plugins
```
I have an `index.php` file in root, which only defines WP_USE_THEMES as `true`, and with `require( 'wordpress/wp-blog-header.php' );`

Then my wp-config.php is in root as well, where I have (besides the other necessary bits):
```
define( 'WP_HOME', 'http://local.domain' );
define( 'WP_SITEURL', WP_HOME . '/wordpress' );
define( 'WP_CONTENT_DIR', realpath( $_SERVER['DOCUMENT_ROOT'] . '/content' ) );
define( 'WP_CONTENT_URL', WP_HOME . '/content' );
```

This is the structure I use for all my sites, and it works. wp-cli usually works as well with this. Just wondering if the scaffolding command wasn't supported in a structure like this? Also, why would it say the theme was created if nothing is actually created? Could it be it is creating the files somewhere else? Couldn't find them anywhere, and I'm not knowlegeable enough to debug the phar file to see on which step something is going amiss.

It's a single site clean install, no plugins. `local.domain` resolves to localhost, obviously.

Thanks in advance for any help, and regards.

WP 4.1.1, OSX Yosemite.

**Comment by yivi** [comment]
In a regular wp install (without putting wp files in a separate folder, and using just wp-content) it does generate the theme files.

**Comment by danielbachhuber** [comment]
Can you check to see if it was created in `wordpress/wp-content/themes` ?

**Comment by yivi** [comment]
That's the first thing I checked :)
No, no theme files in wordpress/wp-content.

Still; it would surprise me, since it in scaffold.php:191 it is using WP_CONTENT_DIR as a location to put the theme in.
![screen shot 2015-03-06 at 18 47 02](https://cloud.githubusercontent.com/assets/1815039/6530435/37417a72-c431-11e4-8fc0-071fb9371f25.png)

**Comment by drrobotnik** [comment]
check the output of $_SERVER['DOCUMENT_ROOT'] when run through cli. Sometimes it's blank or different than when used through http request. It would probably install the theme higher up your server path within a themes folder.

**Comment by yivi** [comment]
`$_SERVER['DOCUMENT_ROOT']` seems to be blank, when running a simple php script through the cli  that does `print_r($SERVER)`

Been looking around higher up in the machine's path, but couldn't find it anywhere.

And why does it work when the structure is the default one (wordpress files in the root directory, and with wp-content among those files)?

**Comment by yivi** [comment]
Since `$_SERVER['DOCUMENT_ROOT']` was blank, my WP_CONTENT_DIR wasn't being set properly.
I changed that define to: `define( 'WP_CONTENT_DIR', dirname( __FILE__ ) . '/content' );` and now it works properly.

Still, for me the main problem was the false report of success. I added this code to scaffold.php to account for this:

```php
		$unzip_result = unzip_file( $tmpfname, $theme_path );
		unlink( $tmpfname );

		if ( TRUE === $unzip_result ) {
			WP_CLI::success( "Created theme '{$data['theme_name']}'." );
		} else {
			WP_CLI::error( "Could not decompress your theme files ('$tmpfname') at '$theme_path': " . $unzip_result->get_error_message() );
			print_r( $unzip_result );
		}
```

It's probably too trivial to merit a pull request, and maybe the error is too verbose for some people? I just thought that it would give the appropriate hints for people like me who would stumble on something like this.
